src/lib/azure/storage.ts

```diff
@@ -0,0 +1,195 @@
+      if (account.name === accountName) {
+        exists = true;
+        break;
+      }
+    }
+
+    // Storage account exists so exit the method.
+    if (exists) {
+      logger.info(`Found ${message}`);
+      return;
+    }
+    logger.info(`${message} does not exist`);
+    // Storage account does not exist so create it.
+    await createStorageAccount(resourceGroup, accountName, location);
+  } catch (err) {
+    logger.error(err);
+  }
+};
+
+/**
+ * Creates Azure storate account `name` in resource group `resourceGroup` in 1ocation `location`
+ *
+ * @param resourceGroup Name of Azure reesource group
+ * @param accountName The Azure storage account name
+ * @param location The Azure storage account location
+ */
+export const createStorageAccount = async (
+  resourceGroup: string,
+  accountName: string,
+  location: string
+) => {
+  const message = `Azure storage account ${accountName} in resource group ${resourceGroup} in ${location} location.`;
+  try {
+    logger.verbose(`Create storage client object.`);
+    const client = await getStorageClient();
+    logger.verbose(
+      `Checking for storage account name ${accountName} availability`
+    );
+    const response: StorageAccountsCheckNameAvailabilityResponse = await client.storageAccounts.checkNameAvailability(
+      accountName
+    );
+
+    if (response.nameAvailable === false) {
+      logger.error(
+        `Storage account name ${accountName} is not available. Please choose a different name.`
+      );
+      return;
+    }
+
+    logger.verbose(`Storage account name ${accountName} is availabile`);
+
+    // Proceed to create a storage account
+    const createParameters = {
+      kind: "Storage" as Kind,
+      location,
+      sku: {
+        name: "Standard_LRS" as SkuName
+      }
+    };
+
+    logger.info(`Creating ${message}`);
+    const resp: StorageAccountsCreateResponse = await client.storageAccounts.create(
+      resourceGroup,
+      accountName,
+      createParameters
+    );
+
+    logger.info(`Created ${message}`);
+  } catch (err) {
+    logger.error(`Error occurred while creating ${message}`);
+    logger.error(err);
+  }
+};
+
+/**
+ * Get storage account `accountName` primary key in resource group `resourceGroup` nd returns the primary key `Promise<string>`
+ *
+ * @param resourceGroup Name of Azure reesource group
+ * @param accountName The Azure storage account name
+ *
+ */
+export const getStorageAccountKey = async (
+  resourceGroup: string,
+  accountName: string
+): Promise<string | undefined> => {
+  try {
+    const client = await getStorageClient();
+    logger.verbose(`Reading storage account ${accountName} access keys`);
+
+    const keyResults: StorageAccountsListKeysResponse = await client.storageAccounts.listKeys(
+      resourceGroup,
+      accountName
+    );
+    if (keyResults.keys === undefined) {
+      logger.verbose(`Storage account ${accountName} access keys do not exist`);
+      return undefined;
+    }
```

Comment:
Should throw error here too, along with `logger.error` instead of `logger.verbose`

Reply:
undefined is valid in the module since storage exist without any access keys, but we can throw error in the calling method.